### PR TITLE
Last refactor

### PR DESCRIPTION
**Pull request by Aferencz1987**
Added sales analyst files and fixed issue in sales engine to make the spec harness work.

### CODE REVIEW

**Review by lcpulzone — COMMENTED** [review]
great refactoring

**Review by aliyamerali — APPROVED** [review]
Looking good!

File: lib/item_repository.rb [hunk 1]
```diff
@@ -1,8 +1,13 @@
+require_relative '../lib/item'
 class ItemRepository
 
   def initialize(parsed_data)
     create_items(parsed_data)
   end
+  #add so spec harness would run successfully.
+  def inspect
+  "#<#{self.class} #{@items_array.size} rows>"
```
[the remark this answers is not in the document]
Comment (matthewjholmes) [reply lib/item_repository.rb:9]: Nice find

**Comment by matthewjholmes** [comment]
Good progress.

**Pull request merged**